camayoc/tests/qcs/test_scan.py

```diff
@@ -31,55 +31,80 @@ def sources():
     return srcs
 
 
-def wait_until_complete(scan, timeout=120):
-    """Wait until the scan has either failed or completed.
+def first_source():
+    """Gather sources from config file."""
+    try:
+        src = [config.get_config()['qcs']['profiles'][0]]
+    except (ConfigFileNotFoundError, KeyError):
+        src = [{'hosts': ['localhost'], 'name':'localhost', 'auths':'default'}]
+    return src
+
+
+def wait_until_state(scan, timeout=120, state='completed'):
+    """Wait until the scan has failed or reached desired state.
+
+    The default state is 'completed'.
 
     This method should not be called on scan jobs that have not yet been
     created, are paused, or are canceled.
 
     The default timeout is set to 120 seconds, but can be overridden if it is
     anticipated that a scan may take longer to complete.
     """
-    if scan.status() in ['paused', 'cancelled']:
+    if scan.status() in ['paused', 'canceled']:
         raise UserWarning(
-            'You have called wait_until_complete() on a scan '
-            'that is paused or cancelled, and this could hang forever.'
+            'You have called wait_until_state() on a scan '
+            'that is paused or canceled, and this could hang forever.'
             ' This exception has been raised instead.'
         )
     while (
         not scan.status() or not scan.status() in [
             'failed',
-            'completed']) and timeout > 0:
+            state]) and timeout > 0:
         time.sleep(5)
         timeout -= 5
```

Comment:
I think if the timeout is reached then an exception should be raised. That will help us identify if a scan was created but never  gone to the running state  and we were expecting it to go, in that case we should not proceed doing any other operation. That is because the scan wasn't on the expected state and the rest of the test can be compromised and generate a false positive.

Reply:
Also with the timeout raising the exception we can use try/except if we are expecting it to timeout and continue the test.